frontend/app/_components/RoleSelection.js

```diff
@@ -6,14 +6,14 @@ const RoleSelection = ({ setRole }) => {
       <h2 className='text-2xl font-bold mb-4 text-gray-800'>Select Your Role</h2>
 
       <button
-        onClick={() => setRole('developer')}
+        onClick={() => setRole('Developer')}
```

Comment:
considering the fact that you have two roles set as a string and don't use Typescript, I would suggest making an enum with the roles. This way you will ensure that there are not bugs related to typos in string or upper/lower case missmatch etc.